Library/Homebrew/brew.sh

```diff
@@ -414,7 +414,8 @@ update-preinstall() {
   [[ -z "$HOMEBREW_AUTO_UPDATE_CHECKED" ]] || return
   [[ -z "$HOMEBREW_UPDATE_PREINSTALL" ]] || return
 
-  if [[ "$HOMEBREW_COMMAND" = "install" || "$HOMEBREW_COMMAND" = "upgrade" || "$HOMEBREW_COMMAND" = "tap" ||
+  if [[ "$HOMEBREW_COMMAND" = "install" || "$HOMEBREW_COMMAND" = "upgrade" ||
+        ( "$HOMEBREW_COMMAND" = "tap" && $HOMEBREW_ARG_COUNT -gt 1 ) ||
```

Comment:
```suggestion
        { "$HOMEBREW_COMMAND" = "tap" && $HOMEBREW_ARG_COUNT -gt 1 } ||
```

[Grouping without invoking a subshell](https://stackoverflow.com/a/29261377/1661012).

Reply:
Thanks for the suggestion, but it gave me a syntax error when I tried it. I've instead just removed the `()` in 45b576c313c6f71b5be54cb86349b3a87f4805dc after confirming that `&&` has higher precedence than `||`, so I think it has the same effect

* http://tldp.org/LDP/abs/html/opprecedence.html

Reply:
I think you need a semicolon in there with the command grouping.